platform/javascript/js/engine/engine.js

```diff
@@ -274,5 +269,5 @@ const Engine = (function() {
 	Engine.prototype['setPersistentPaths'] = Engine.prototype.setPersistentPaths;
 	Engine.prototype['requestQuit'] = Engine.prototype.requestQuit;
 	return Engine;
-})();
+}());
```

Comment:
Is that [JSFuck](https://en.wikipedia.org/wiki/JSFuck)? ;)

Reply:
Eheh, this is JSBeuty :).
so... the best thing in JavaScript, the thing I really really like about it, are closures.
People may think JavaScript cannot have private members, while in reality, you can have that using closures.
So, say I have a constructor function:
```javascript
function Company() {
    var member = 1;
    this.addMember = function() {
        return ++member;
    }
}
var theCompany = new Company();
theCompany.addMember() // will print 2
theCompany.addMember() // will print 3
```

Now imagine you want `theCompany` to be a singleton, and actually disallow creating a new company.
We can create an anonymous factory function, and run it immediately:
```javascript
var theCompany = function() { // An anonymous function
	// Same code as above
	function Company() {
	    var member = 1;
	    this.addMember = function() {
		return ++member;
	    }
	}
        return new Company(); // Return the singleton
}/* end of function */()/* call the anonymous function */
```
I think the extra wrapping around `function` could be removed in this case, i.e. `(function(){}())` could just be `function(){}()`, which is why the linter switched the order of the parenthesis, wrapping the outside instead of the inside.